financial-connections/src/main/java/com/stripe/android/financialconnections/domain/NativeAuthFlowRouter.kt

```diff
@@ -10,10 +11,12 @@ import javax.inject.Inject
  * Additionally, handles logging exposures when needed.
  */
 internal class NativeAuthFlowRouter @Inject constructor(
-    val eventTracker: FinancialConnectionsAnalyticsTracker
+    private val eventTracker: FinancialConnectionsAnalyticsTracker,
+    private val debugConfiguration: DebugConfiguration
 ) {
 
     fun nativeAuthFlowEnabled(sync: SynchronizeSessionResponse): Boolean {
+        debugConfiguration.overridenNative?.let { return it }
```

Comment:
@vardges-stripe small update to the `NativeAuthFlowRouter` - debug configuration is checked before running the actual router logic.

- DebugConfiguration has a mock implementation in release builds - https://github.com/stripe/stripe-android/pull/5863/files#diff-bdc565fc2134f9f5ee84ae86f5c5b62dbf5c748b359e89ed58f18a55589e6881
- ... And a real configuration reading from shared prefs (Android's `NSUserDefaults`) in debug builds.